beacon-chain/execution/log_processing.go

```diff
@@ -221,6 +222,19 @@ func (s *Service) ProcessDepositLog(ctx context.Context, depositLog gethtypes.Lo
 			"merkleTreeIndex": index,
 		}).Info("Invalid deposit registered in deposit contract")
 	}
+	if features.Get().EnableEIP4881 {
+		// We finalize the trie here so that old deposits are not kept around, as they make
+		// deposit tree htr computation expensive.
+		dTrie, ok := s.depositTrie.(*depositsnapshot.DepositTree)
+		if !ok {
+			log.Errorf("Wrong trie type initialized: %T", dTrie)
+			return nil
```

Comment:
This should probably return an error too? This seems like an issue that would make the client unstable so that error should be surfaced up the call stack.